### ISSUE TITLE
Discussion: strategic call linkage redesign

### ISSUE THREAD

**Issue by zl-wang** [eclipse/omr#4493]
@0xdaryl looked for s strategic redesign of OMR calling linkage.  Although it is unclear yet what exactly constitutes a strategic redesign, it might need to include at least:
1) Decoupling call-linkage from the underlying symbol;
2) Derived the linkage from the call Node, or Node-specific linkage field itself;

I am inviting discussions from:  @0xdaryl @andrewcraik @fjeremic @vijaysun-omr @mstoodle @gita-omr

**Comment by andrewcraik** [comment]
I don't think it makes sense to store the linkage on the Node - the linkage is either a property of the symbol or symbol reference in my mind - nodes shouldn't normally hold this kind of state IMO.
